utilitiesApp/src/dbLoadRecordsFuncs.cpp

```diff
@@ -66,26 +67,18 @@ static void loadMacEnviron(MAC_HANDLE* pmh)
 	}
 }
 
-/// look for e.g. \$(I) and replace with $(I) so we can substitute later with macEnvExpand()
+/// look for e.g. \$() and replace with $() so we can substitute later with macEnvExpand()
 static void subMacros(std::string& new_macros, const char* macros, const char* loopVar)
 {
-    char loopSubFrom[32], loopSubTo[32];
-    epicsSnprintf(loopSubFrom, sizeof(loopSubFrom), "\\$(%s)", loopVar);
-    epicsSnprintf(loopSubTo, sizeof(loopSubTo), "$(%s)", loopVar);
     new_macros = macros;
-    size_t start_pos = 0, lf = strlen(loopSubFrom), lt = strlen(loopSubTo);
-    while( (start_pos = new_macros.find(loopSubFrom, start_pos)) != std::string::npos ) 
-    {
-        new_macros.replace(start_pos, lf, loopSubTo);
-        start_pos += lt;
-    }
-    epicsSnprintf(loopSubFrom, sizeof(loopSubFrom), "\\${%s}", loopVar);
-    epicsSnprintf(loopSubTo, sizeof(loopSubTo), "${%s}", loopVar);
-	start_pos = 0;
-    while( (start_pos = new_macros.find(loopSubFrom, start_pos)) != std::string::npos ) 
-    {
-        new_macros.replace(start_pos, lf, loopSubTo);
-        start_pos += lt;
+    std::smatch bracketSearch, braceSearch;
+    // Check that we have done a search, and that the result of both is empty before finishing.
+    while((!bracketSearch.empty())||(!braceSearch.empty())||(!bracketSearch.ready())){
+        std::regex_search(new_macros, bracketSearch, std::regex(R"(\\\$\(([0-9a-zA-Z_\$\(\){}]+)\))"));
+        new_macros = std::regex_replace(new_macros, std::regex(R"(\\\$\(([0-9a-zA-Z_\$\(\){}]+)\))"), "$($1)");
```

Comment:
The `std::regex` expression is repeated, could this be taken out into a constant? Also I believe you can skip the `\` escape within a `[` construct as it knows you just want normal characters i.e.    `[0-9a-zA-Z_$(){}]`